service/frontend/workflow_handler.go

```diff
@@ -3226,22 +3226,21 @@ func (wh *WorkflowHandler) ListSchedules(
 		return nil, errListNotAllowed
 	}
 
+	query := ""
+	if strings.TrimSpace(request.Query) != "" {
+		query = fmt.Sprintf("%s AND (%s)", scheduler.VisibilityBaseListQuery, request.Query)
```

Comment:
We can't just use the query as-is: that will allow filtering on all workflow search attributes, e.g `ExecutionStatus`. But schedules don't have `ExecutionStatus`, they have only `TemporalSchedulePaused` and custom search attributes. We need to limit this query to only those.

(In other words: this api deals with durable entities called schedules. This current implementation chooses to implement them with workflows, but the semantics of workflows and schedules are not the same. This leaks details about the implementation and may cause problems when we change the implementation.)